common/tasks/interleaved_weighted_round_robin.go

```diff
@@ -439,49 +459,69 @@ LoopDispatch:
 			continue LoopDispatch
 		}
 
-		now := s.timeSource.Now()
-		reservation := rateLimiter.Reserve(
-			now,
-			channel.rateLimitRequest,
-		)
-		if reservation.DelayFrom(now) != 0 {
-			reservation.CancelAt(now)
-			channel.throttled = true
-			numThrottled++
-			if numThrottled == len(iwrrChannels.channels) {
-				// all channels throttled
-				break LoopDispatch
-			}
+		if len(channel.Chan()) == 0 {
 			continue LoopDispatch
 		}
+
+		if !rateLimiter.Allow(s.timeSource.Now(), channel.rateLimitRequest) {
+			s.metricsHandler.Counter(metrics.TaskSchedulerThrottled.GetMetricName()).Record(1, channel.metricsTags...)
+
+			if rateLimiterConfig.enableThrottle {
+				channel.throttled = true
+				numThrottled++
+				if numThrottled == len(iwrrChannels.channels) {
+					// all channels throttled
+					break LoopDispatch
+				}
+				continue LoopDispatch
```

Comment:
Can we make .Wait() instead of .Allow() to prevent looping here?

Reply:
Synced offline. At least right now there's no easy way.